app/models/vote.rb

```diff
@@ -0,0 +1,11 @@
+class Vote < ApplicationRecord
+  belongs_to :user
+  belongs_to :work
+
+  def unique_vote
+    unless !Vote.all.any?{|v| v.user_id == user_id && v.work_id == work_id}
+      errors[:name] << "You have already voted!"
+    end
+  end
+  validate :unique_vote
```

Comment:
Nice work figuring out the custom validation! You could also have used the built-in `uniqueness` validation:

```
validates :work_id, uniqueness: { scope: :user_id }
```